docs/design/dab-validate.md

```diff
@@ -0,0 +1,53 @@
+# Design Specification: Validating a Config File
+
+## Overview
+
+This design specification outlines the process for validating the Data API builder config file used by a runtime engine. The validation process ensures that the config file is correctly formatted and contains all the required information for the runtime engine to function correctly.
+
+## Requirements
+
+The following requirements must be met by the validation process:
+
+- The validation process must check that the configuration file is correctly formatted and is in sync with the schema file.
+- If the configuration file is not valid, the validation process must 
+    - throw an exception in case of Metadata Initialization failure with a descriptive error message.
+    - Provide a list of errors that occur when the configuration file does not conform to the schema, including the line numbers where each error occurs.
```

Comment:
How doable is this? I'd imagine that the first error encountered results in an exception and log event. are you just going to continue processing so that all errors can be emitted in one pass through?

Reply:
yes, instead of throwing the error we will collect it, but some places we would still throw the error like if the connection string is not valid, we won't continue to fetch entity metadata from the database.